source/agora/node/FullNode.d

```diff
@@ -648,7 +674,29 @@ public class FullNode : API
 
     protected TransactionPool getPool ()
     {
-        return new TransactionPool(this.cacheDB);
+        return new TransactionPool(this.cacheDB, getDoubleSpentSelector());
+    }
+
+    /***************************************************************************
+
+        Returns an instance of a DoubleSpentSelector
+
+        Subclasses can override this method and return
+        a DoubleSpentSelector
+
+        Returns:
+            an instance of DoubleSpentSelector
+
+    ***************************************************************************/
+
+    protected DoubleSpentSelector getDoubleSpentSelector () nothrow @safe
```

Comment:
Why not make this function the double spend detector (and passing its address) instead of the extra dg ?

Reply:
I like that idea...